server/src/main/java/org/opensearch/search/internal/ContextIndexSearcher.java

```diff
@@ -389,11 +395,44 @@ protected void searchLeaf(LeafReaderContext ctx, int minDocId, int maxDocId, Wei
             }
         }
 
+        if (searchContext.isStreamSearch()) {
+            logger.debug(
+                "Stream intermediate aggregation for segment [{}], shard [{}]",
+                ctx.ord,
+                searchContext.shardTarget().getShardId().id()
+            );
+            List<InternalAggregation> internalAggregation = searchContext.bucketCollectorProcessor().buildAggBatch(collector);
+            if (!internalAggregation.isEmpty()) {
+                sendBatch(internalAggregation);
+            }
+        }
+
         // Note: this is called if collection ran successfully, including the above special cases of
         // CollectionTerminatedException and TimeExceededException, but no other exception.
         leafCollector.finish();
     }
 
+    public void sendBatch(List<InternalAggregation> batch) {
```

Comment:
this can be package-private, its only used in tests and locally.

Reply:
yeah, ContextIndexSearch is annotated with PublicAPI, it's bad to add public method in here.
updated